test/index.js

```diff
@@ -21,334 +21,275 @@ const expect = Code.expect;
-            });
-        });
+        try {
+            await client.set(key, value, 10);
+        }
+        catch (err) {
+            expect(err).to.exist();
+        }
     });
 
-    it('ignored starting a connection twice on same event', (done) => {
+    it('ignored starting a connection twice on same event', async () => {
 
         const client = new Catbox.Client(Mongo);
         let x = 2;
-        const start = function () {
-
-            client.start((err) => {
+        const start = async function () {
 
-                expect(err).to.not.exist();
-                expect(client.isReady()).to.equal(true);
-                --x;
-                if (!x) {
-                    done();
-                }
-            });
+            await client.start();
+            expect(client.isReady()).to.equal(true);
+            --x;
+            if (!x) {
+            }
         };
 
-        start();
-        start();
+        await start();
+        await start();
     });
 
-    it('ignored starting a connection twice chained', (done) => {
+    it('ignored starting a connection twice chained', async () => {
 
         const client = new Catbox.Client(Mongo);
-        client.start((err) => {
+        await client.start();
+        expect(client.isReady()).to.equal(true);
 
-            expect(err).to.not.exist();
-            expect(client.isReady()).to.equal(true);
-
-            client.start((err) => {
-
-                expect(err).to.not.exist();
-                expect(client.isReady()).to.equal(true);
-                done();
-            });
-        });
+        await client.start();
+        expect(client.isReady()).to.equal(true);
     });
 
-    it('returns not found on get when using null key', (done) => {
+    it('returns not found on get when using null key', async () => {
 
         const client = new Catbox.Client(Mongo);
-        client.start((err) => {
+        await client.start();
 
-            expect(err).to.not.exist();
-            client.get(null, (err, result) => {
-
-                expect(err).to.equal(null);
-                expect(result).to.equal(null);
-                done();
-            });
-        });
+        const result = await client.get(null);
+        expect(result).to.equal(null);
     });
 
-    it('returns not found on get when item expired', (done) => {
+    it('returns not found on get when item expired', async () => {
 
         const client = new Catbox.Client(Mongo);
-        client.start((err) => {
+        await client.start();
 
+        const key = { id: 'x', segment: 'test' };
+        try {
+            await client.set(key, 'x', 1);
+        }
+        catch (err) {
             expect(err).to.not.exist();
-            const key = { id: 'x', segment: 'test' };
-            client.set(key, 'x', 1, (err) => {
-
-                expect(err).to.not.exist();
-                setTimeout(() => {
-
-                    client.get(key, (err, result) => {
+            setTimeout(async () => {
```

Comment:
I suspect that the lab won't wait until this setTimeout is evaluated. Since test function basically ends when returned promise gets resolved.

Reply:
To make these patterns nicely consistent within the hapi ecosystem - its often useful to look at how its approached in hapi core module. For example in this case - you can see one how the setTimeout can be used in async function - https://github.com/hapijs/hapi/blob/286324dcd61733148e3759f325659eeffe124184/test/request.js#L265

Reply:
That, or `hoek.wait`.